armbotoffboard/examplesmartmotorcontroller.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) FIRST and other WPILib contributors.
+# Open Source Software; you can modify and/or share it under the terms of
+# the WPILib BSD license file in the root directory of this project.
+
+import wpilib.interfaces
+import enum
+
+
+class ExampleSmartMotorController(wpilib.interfaces.MotorController):
+    """A simplified stub class that simulates the API of a common "smart" motor controller.
+    Has no actual functionality.
+    """
+
+    class PIDMode(enum.Enum):
+        kPosition = enum.auto()
+        kVelocity = enum.auto()
+        kMovementWitchcraft = enum.auto()
+
+    def __init__(self, port: int) -> None:
+        """Creates a new ExampleSmartMotorController.
+
+        Args:
+            port: The port for the controller.
+        """
+        super().__init__()
+
+    def setPID(self, kp: float, ki: float, kd: float) -> None:
+        """Example method for setting the PID gains of the smart controller.
+
+        Args:
+            kp: The proportional gain.
+            ki: The integral gain.
+            kd: The derivative gain.
+        """
+        pass
+
+    def setSetPoint(mode: PIDMode, setpoint: float, arbfeedforward: float) -> None:
+        """Example method for setting the setpoint of the smart controller in PID mode.
+
+        Args:
+            mode: The mode of the PID controller.
+            setpoint: The controller setpoint.
+            arbfeedforward: An arbitrary feedforward output (from -1 to 1).
+        """
+        pass
+
+    def follow(self, leader: __qualname__) -> None:
```

Comment:
```suggestion
    def follow(self, leader: "ExampleSmartMotorController") -> None:
```
I don't believe PEP 484 says that type checkers should understand `__qualname__` in type hints.